src/%IPM/ResourceProcessor/Abstract.cls

```diff
@@ -225,35 +225,47 @@ Method OnGetUniqueName(Output pUniqueName)
 {
 }
 
-Method GetSourceControlInfo(Output pInfo As %IPM.PackageManager.Developer.Extension.SourceControl.ResourceInfo) As %Status
+Method GetSourceControlInfo(Output pInfo As %IPM.ExtensionBase.SourceControl.ResourceInfo) As %Status
 {
-	Set pInfo = ##class(%IPM.PackageManager.Developer.Extension.SourceControl.ResourceInfo).%New()
+	Set pInfo = ##class(%IPM.ExtensionBase.SourceControl.ResourceInfo).%New()
 	Quit $$$OK
 }
 
-/// Evaluates an expression in an attribute
+
+/// Evaluates an expression in a provided string. <br />
+/// These special expressions are case-insensitive. <br />
 /// Current valid expressions:
-/// {$namespace} - the current namespace
-/// {$mgrdir} - the instance's mgr directory
-/// {$cspdir} - the instance's root CSP directory
-/// {$root} - the resource's module's root directory
-/// {name} - the name of the resource
-/// These special expressions are case-insensitive.
-Method %Evaluate(pAttrValue) As %String [ Internal ]
+/// <ul>
+/// 	<li>{$root} - the resource's module's root directory </li>
+/// 	<li>{name} - the name of the resource </li>
+/// </ul>
+/// In addition to these, look at expressions supported by %EvaluateSystemExpression
+/// in class <class>%IPM.Utils.Module</class>.
+Method %Evaluate(pAttrValue As %String) As %String [ Internal ]
 {
 	Set tAttrValue = pAttrValue
 	If $Data(..Params) {
 		Merge tParams = ..Params
 		Do ##class(%IPM.Storage.ModuleSetting.Default).GetCustomParameters(.customParams,..ResourceReference.Module, .tParams)
 		Set tAttrValue = ##class(%IPM.Storage.ModuleSetting.Default).EvaluateAttribute(tAttrValue,.customParams)
 	}
-
-	If $Locate(tAttrValue,"(?i)\{name\}") {
-		Set tName = ..ResourceReference.Name
-   	Set tAttrValue = ##class(%Regex.Matcher).%New("(?i)\{name\}", tAttrValue).ReplaceAll($Replace(tName, "\","\\"))
+	If (tAttrValue '[ "{") {
+		Return tAttrValue
 	}
-
-  Quit ..ResourceReference.Module.%Evaluate(tAttrValue)
+	Set attrValue = ##class(%IPM.Utils.Module).%EvaluateSystemExpression(tAttrValue)
+	// for backward compatibility, supporting both {$root} and ${root}
+	If $Locate(attrValue,"(?i)\{\$root\}|\$\{root\}") {
+		Set root = ..ResourceReference.Module.Root
+		If (root '= "") {
+			Set root = ##class(%Library.File).NormalizeDirectory(root)
+		}
+		Set attrValue = ##class(%IPM.Utils.Module).%RegExReplace(attrValue,"(?i)\{\$root\}|\$\{root\}",root)
```

Comment:
no need in $locate, and with fix in RegExReplace, it will be just `%RegExReplace(attrValue, "root",root)`